### PR TITLE
Adding graph update mode for merge index image API endpoint

### PR DESCRIPTION
**Pull request by lipoja**
[CLOUDDST-20163]

### CODE REVIEW

File: iib/web/iib_static_types.py [hunk 1]
```diff
@@ -388,6 +389,7 @@ class MergeIndexImageRequestResponse(APIPartImageBuildRequestResponse):
     build_tags: List[str]
     deprecation_list: List[str]
     distribution_scope: str
+    graph_update_mode: str
```
Comment (xDaile) [inline iib/web/iib_static_types.py:392]: Type can be Literal["replaces", "semver", "semver-skippatch"]

File: iib/workers/tasks/build_merge_index_image.py [hunk 2]
```diff
@@ -179,6 +184,7 @@ def handle_merge_request(
     distribution_scope: Optional[str] = None,
     binary_image_config: Optional[str] = None,
     build_tags: Optional[List[str]] = None,
+    graph_update_mode: Optional[str] = None,
```
Comment (xDaile) [inline iib/workers/tasks/build_merge_index_image.py:187]: I suppose Mypy could have a problem with this line (using Literal and str].

**Review by xDaile — DISMISSED** [review]

File: iib/web/iib_static_types.py [hunk 3]
```diff
@@ -390,7 +390,7 @@ class MergeIndexImageRequestResponse(APIPartImageBuildRequestResponse):
     build_tags: List[str]
     deprecation_list: List[str]
     distribution_scope: str
-    graph_update_mode: Literal['replaces', 'semver', 'semver-skippatch']
+    graph_update_mode: GRAPH_MODE_LITERAL
```
Comment (JAVGan) [inline iib/web/iib_static_types.py:393]: Is it possible to do this in the previous commit instead of this one?

File: iib/web/iib_static_types.py [hunk 4]
```diff
@@ -162,7 +162,7 @@ class MergeIndexImagesPayload(TypedDict):
     build_tags: NotRequired[List[str]]
     deprecation_list: NotRequired[List[str]]
     distribution_scope: NotRequired[str]
-    graph_update_mode: NotRequired[str]
+    graph_update_mode: NotRequired[GRAPH_MODE_LITERAL]
```
Comment (JAVGan) [inline iib/web/iib_static_types.py:165]: Is it possible to do this in the previous commit instead of this one?

**Review by JAVGan — APPROVED** [review]
Besides the fix for the first commit in the second one everything else LGTM

File: iib/web/iib_static_types.py (hunk 4 above)
Comment (lipoja) [reply iib/web/iib_static_types.py]: The other way around. The `graph_update_mode` should not be present in the first commit.

File: iib/web/iib_static_types.py (hunk 3 above)
Comment (lipoja) [reply iib/web/iib_static_types.py]: The other way around. The graph_update_mode should not be present in the first commit.

**Review by JAVGan — APPROVED** [review]

**Review by xDaile — APPROVED** [review]

**Review by chandwanitulsi — APPROVED** [review]

**Pull request merged**